Cleanup cli

jgbradley1 opened the pull request:
## Description

This PR cleans up and standardizes formatting of the various CLI. It also adds safeguards (file and directory existence checks) where appropriate.

There was one bug discovered in the `prompt_tune` module where if `--config` and `--root` were specified, the correct filepath to the config file would not get used and error out.

## Checklist

- [x] I have tested these changes locally.
- [x] I have reviewed the code changes.
- [x] I have updated the documentation (if necessary).
- [x] I have added appropriate unit tests (if applicable).

andresmor-ms commented on graphrag/index/progress/load_progress_reporter.py:
```diff
@@ -19,12 +19,13 @@ def load_progress_reporter(reporter_type: str = "none") -> ProgressReporter:
-    msg = f"Invalid progress reporter type: {reporter_type}"
-    raise ValueError(msg)
+    match reporter_type:
+        case "rich":
```
You could use the ReporterType enum in here instead of strings

jgbradley1 replied:
Done

andresmor-ms commented on graphrag/prompt_tune/__main__.py:
```diff
@@ -5,126 +5,116 @@
 from .loader import MIN_CHUNK_SIZE
 
+
+def file_exist(path):
+    """Check for file existence."""
+    if not Path(path).is_file():
+        msg = f"File not found: {path}"
+        raise argparse.ArgumentTypeError(msg)
+    return path
+
+
+def dir_exist(path):
+    """Check for directory existence."""
+    if not Path(path).is_dir():
+        msg = f"Directory not found: {path}"
+        raise argparse.ArgumentTypeError(msg)
+    return path
```
You could move these functions into a utils to avoid duplication of code

jgbradley1 replied:
Done. Moved to the top-level utils directory.

AlonsoGuevara commented on graphrag/index/__main__.py:
```diff
@@ -4,73 +4,123 @@
+class ReporterType(Enum):
+    """The type of reporter to use."""
+
+    RICH = "rich"
```
I've always wanted to replace our enum strings with auto(), would make the implementation cleaner.
WDYT? Not suggesting this as a refactor, but a conversation starter.

AlonsoGuevara replied:
BTW, fully aware this is a cli PR :) I'm speaking in general hehe

jgbradley1 replied:
Sounds like a good idea initially however I think we need the string representation for the CLI. I don't think `argparse` can display/handle enums very cleanly without an overloaded `__str__` or requiring more code.

For internal library operations, `auto` would be cleaner but then we're maintaining two different styles of enums which is worse in my opinion ☹️.

AlonsoGuevara commented on graphrag/index/__main__.py:
```diff
@@ -79,20 +129,20 @@
     args = parser.parse_args()
 
     if args.resume and args.update_index:
-        msg = "Cannot resume and update a run at the same time."
```
Interesting. Previously Ruff failed when there wasn't a dot. Guess it now changed :)

AlonsoGuevara commented on graphrag/index/api.py:
```diff
@@ -22,12 +22,12 @@
 
 async def build_index(
     config: GraphRagConfig,
+    emit: list[str],
```
do you think it makes sense, at least from a conceptual standpoint, to have this param so high in the signature?

jgbradley1 replied:
I originally moved this due to removing the default value of `None`. It's been moved back though and the default value set to `parquet`. At the API layer, I think this makes more sense.

We don't want the defualt value to be defined as `None` at the API layer because that gets changed to `parquet` inside the library anyway. It's best to make this default behavior more transparent.

AlonsoGuevara commented on graphrag/index/__main__.py:
```diff
@@ -4,73 +4,123 @@
+        return self.value
+
+
+def file_exist(path):
```
THis looks like it could go into the top level utils module

jgbradley1 replied:
Done. Moved to the top-level utils directory.

AlonsoGuevara commented on graphrag/query/__main__.py:
```diff
@@ -22,69 +23,74 @@ def __str__(self):
         return self.value
 
 
+def file_exist(path):
```
yup, top level utils :)

jgbradley1 replied:
Done. Moved to the top-level utils directory.

jgbradley1 commented:
@AlonsoGuevara / @andresmor-ms I moved those functions into the `utils` directory. and incorporated more use of some existing Enum classes.

Let me know if there is any other feedback. Otherwise I think this is ready to merge.
